### PR TITLE
nginx: add luci-support package

### PR DESCRIPTION
**Pull request by Ansuel**
This adds an additional file for ngix that contains all the files need to make luci works on the nginx webserver.

@hnyman i think there is a lot to change... starting from the uci-defaults scripts...
Should i use a post install script? 

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

### CODE REVIEW

File: net/nginx/Makefile [hunk 1]
```diff
@@ -209,6 +209,32 @@ ifeq ($(CONFIG_NGINX_HTTP_SECURE_LINK),y)
...
+define Package/nginx-mod-luci
+  SECTION:=net
+  CATEGORY:=Network
+  SUBMENU:=Web Servers/Proxies
+  TITLE:=Support file for Nginx
+  URL:=http://nginx.org/
+  DEPENDS:=+nginx +uwsgi-cgi
+endef
+
+define Package/nginx-mod-LuCI/description
```
Comment (hnyman) [inline net/nginx/Makefile]: description has different spelling in package name.
Might not work.

File: net/nginx/Makefile [hunk 2]
```diff
@@ -209,6 +209,32 @@ ifeq ($(CONFIG_NGINX_HTTP_SECURE_LINK),y)
...
+define Package/nginx-mod-luci-ssl
+  SECTION:=net
+  CATEGORY:=Network
+  SUBMENU:=Web Servers/Proxies
+  TITLE:=Support file for Nginx (SSL)
+  URL:=http://nginx.org/
+  DEPENDS:=+nginx-mod-luci
+endef
+
+define Package/nginx-mod-LuCI/description
```
Comment (hnyman) [inline net/nginx/Makefile]: different spelling

File: net/nginx/files-luci-support/uwsgi.init [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#!/bin/sh /etc/rc.common
```
Comment (hnyman) [inline net/nginx/files-luci-support/uwsgi.init]: uwsgi init script should be in that package.
you may need also two variants there.

**Comment by Ansuel** [comment]
what about uci-defaults package?

File: net/nginx/Makefile [hunk 4]
```diff
@@ -218,6 +217,32 @@ ifeq ($(CONFIG_NGINX_HTTP_BROTLI),y)
...
+define Package/nginx-mod-luci-ssl
+  SECTION:=net
+  CATEGORY:=Network
+  SUBMENU:=Web Servers/Proxies
+  TITLE:=Support file for Nginx (SSL)
+  URL:=http://nginx.org/
+  DEPENDS:=nginx +nginx-mod-luci
+endef
+
+define Package/nginx-mod-luci/description
```
Comment (hnyman) [inline net/nginx/Makefile]: should be nginx-mod-luci-ssl, I think

**Comment by Ansuel** [comment]
yes you are right, anyway i can't underestand when the uci-defaults scripts are used. Opkf execute them after install or only in a clean build condition ?

**Comment by hnyman** [comment]
> i can't underestand when the uci-defaults scripts are used. Opkf execute them after install or only in a clean build condition ?

They supposed to get executed in the live router wither at opkg installation to a running system or at the next boot at the latest (if the package was installed into image). Earlier they were executed only at boot, but in 2016 it was changed to be also done at opkg installation.

**Comment by hnyman** [comment]
Not quite sure if the uci-.defaults way is practical, or would it be better to define two variants like
* nginx/default definition with all other stuff except config
* "nginx" with normal config + reference to nginx/default
* "nginx-luci" with luci config + reference to nginx/default

(similar duplication-avoiding structure that you already did for ariang)

also the service restart/start functionality in uci-defaults looks abnormal. uci-defaults is mostly meant for cnfgi changes, not forcing packages to start.

But lets merge this, so that you can get feedback from the possible users (in the forum thread).

**Pull request merged**